src/main/java/org/takes/rq/RqHref.java

```diff
@@ -67,16 +79,36 @@
         public Base(final Request req) {
             super(req);
         }
+
+        // @todo #445:30min/DEV RqMethod already validates Request-Line and
+        // extracts HTTP Method from it. We should extract all important
+        // information from Request-Line (HTTP method, URI and HTTP version)
+        // in one place to enforce DRY principle.
         @Override
         public Href href() throws IOException {
+            if (!this.head().iterator().hasNext()) {
+                throw new HttpException(
+                    HttpURLConnection.HTTP_BAD_REQUEST,
+                    "HTTP Request should have Request-Line"
+                );
+            }
+            final String requestLine = this.head().iterator().next();
+            final Matcher uriMatcher =
```

Comment:
@Happy-Neko why not just `matcher`? see http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html